src/main/telemetry/esc_telemetry.c

```diff
@@ -0,0 +1,298 @@
+{
+    // KISS ESC sends some data during startup, ignore this for now (maybe future use)
+    // startup data could be firmware version and serialnumber
+
+    if (escTelemetryTriggerState == ESC_TLM_TRIGGER_WAIT) return;
+
+    tlm[tlmFramePosition] = (uint8_t)c;
+
+    if (tlmFramePosition == ESC_TLM_BUFFSIZE - 1) {
+        tlmFrameDone = true;
+        tlmFramePosition = 0;
+    } else {
+        tlmFramePosition++;
+    }
+}
+
+uint8_t escTelemetryFrameStatus(void)
+{
+    uint8_t frameStatus = ESC_TLM_FRAME_PENDING;
+    uint16_t chksum, tlmsum;
+
+    if (!tlmFrameDone) {
+        return frameStatus;
+    }
+
+    tlmFrameDone = false;
+
+    // Get CRC8 checksum
+    chksum = get_crc8(tlm, ESC_TLM_BUFFSIZE - 1);
+    tlmsum = tlm[ESC_TLM_BUFFSIZE - 1];     // last byte contains CRC value
+
+    if (chksum == tlmsum) {
+        escTelemetryData[escTelemetryMotor].skipped = false;
+        escTelemetryData[escTelemetryMotor].temperature = tlm[0];
+        escTelemetryData[escTelemetryMotor].voltage = tlm[1] << 8 | tlm[2];
+        escTelemetryData[escTelemetryMotor].current = tlm[3] << 8 | tlm[4];
+        escTelemetryData[escTelemetryMotor].consumption = tlm[5] << 8 | tlm[6];
+        escTelemetryData[escTelemetryMotor].rpm = tlm[7] << 8 | tlm[8];
+
+        frameStatus = ESC_TLM_FRAME_COMPLETE;
+    }
+
+    return frameStatus;
+}
+
+void escTelemetryProcess(uint32_t currentTime)
+{
+    UNUSED(currentTime);
+
+    if (!escTelemetryEnabled) {
+        return;
+    }
+
+    // Wait period of time before requesting telemetry (let the system boot first)
+    if (millis() < ESC_BOOTTIME)
+    {
+        return;
+    }
+    else if (escTelemetryTriggerState == ESC_TLM_TRIGGER_WAIT)
+    {
+        // Ready for starting requesting telemetry
+        escTelemetryTriggerState = ESC_TLM_TRIGGER_READY;
+        escTelemetryMotor = 0;
+        escTriggerTimestamp = millis();
+    }
+    else if (escTelemetryTriggerState == ESC_TLM_TRIGGER_READY)
+    {
+        if (debugMode == DEBUG_ESC_TELEMETRY) debug[0] = escTelemetryMotor+1;
+
+        motorDmaOutput_t * const motor = getMotorDmaOutput(escTelemetryMotor);
+        motor->requestTelemetry = true;
+        escTelemetryTriggerState = ESC_TLM_TRIGGER_PENDING;
+    }
+
+    if (escTriggerTimestamp + ESC_REQUEST_TIMEOUT < millis())
+    {
+        // ESC did not repond in time, skip to next motor
+        escTelemetryData[escTelemetryMotor].skipped = true;
+        selectNextMotor();
+        escTelemetryTriggerState = ESC_TLM_TRIGGER_READY;
+
+        if (debugMode == DEBUG_ESC_TELEMETRY) debug[1]++;
+    }
+
+    // Get received frame status
+    uint8_t state = escTelemetryFrameStatus();
+
+    if (state == ESC_TLM_FRAME_COMPLETE)
+    {
+        // Wait until all ESCs are processed
+        if (firstCycleComplete)
+        {
+            uint8_t i;
+            escCurrent = 0;
+            escConsumption = 0;
+            for (i = 0; i < 4; i++)             // Motor count for Dshot limited to 4
```

Comment:
and it can be defined inside the for

e.g. `for (int i = 0; i < MAX_SUPPORTED_MOTORS; i++) {` (note the brace on the same line)